admin/spam/views.py

```diff
@@ -1,96 +1,124 @@
 from __future__ import unicode_literals
 
 from django.shortcuts import render
-from django.contrib.auth.decorators import login_required
 from django.core.mail import send_mail
-from django.http import HttpResponseNotFound
 from django.views.generic import FormView, ListView
-from django.utils.decorators import method_decorator
-from django.core.urlresolvers import reverse
 from django.views.defaults import page_not_found
 
 from modularodm import Q
 from website.project.model import Comment
 from website.settings import SUPPORT_EMAIL
 
-from admin.common_auth.logs import update_admin_log, CONFIRM_HAM, CONFIRM_SPAM
+from admin.base.utils import OSFAdmin
+from admin.common_auth.logs import (
+    update_admin_log,
+    CONFIRM_HAM,
+    CONFIRM_SPAM,
+    USER_EMAILED,
+)
 from admin.spam.serializers import serialize_comment
 from admin.spam.forms import EmailForm, ConfirmForm
+from admin.spam.templatetags.spam_extras import reverse_spam_detail
 
+STATUS = dict(
+    UNKNOWN=Comment.UNKNOWN,
+    SPAM=Comment.SPAM,
+    HAM=Comment.HAM,
+    FLAGGED=Comment.FLAGGED,
+)
 
-class EmailFormView(FormView):
 
+class EmailFormView(OSFAdmin, FormView):
+    """ Allow authorized admin user to Email spammers, use judiciously
+
+    Gets user data and supplies email template. Logs mail sent to user.
+    """
     form_class = EmailForm
     template_name = "spam/email.html"
-    spam_id = None
-    page = 1
-
-    def __init__(self):
-        self.spam = None
-        super(EmailFormView, self).__init__()
 
     def get(self, request, *args, **kwargs):
-        spam_id = kwargs.get('spam_id', None)
-        self.spam_id = spam_id
-        self.page = request.GET.get('page', 1)
         try:
-            self.spam = serialize_comment(Comment.load(spam_id))
-        except (AttributeError, TypeError):
-            return HttpResponseNotFound(
-                '<h1>Spam comment ({}) not found.</h1>'.format(spam_id)
+            return super(EmailFormView, self).get(request, *args, **kwargs)
+        except AttributeError:
+            return page_not_found(
```

Comment:
I think you want to [return an `HttpResponseNotFound` and let django call `page_not_found`](https://docs.djangoproject.com/en/1.9/topics/http/views/#returning-errors). That gives you the ability to [set a custom 404 view method](https://docs.djangoproject.com/en/1.9/topics/http/views/#customizing-error-views) in your urls file at any point in the future and all of your 404s will call that method instead of having to go track them all down and replace them.